xcode/Subconscious/Shared/Components/Common/Forms/ValidatedTextField.swift

```diff
@@ -6,39 +6,47 @@
 //
 
 import SwiftUI
+import Combine
 
 /// A text field that comes with help text and a validation flag
 struct ValidatedTextField: View {
     var placeholder: String
     @Binding var text: String
+    var onFocusChanged: ((Bool) -> Void)?
     var caption: String
-    var isValid: Bool = true
-
+    var hasError: Bool = false
+    @FocusState var focused: Bool
```

Comment:
Despite technically being a two-way binding, this is only used to subscribe to the focus value.